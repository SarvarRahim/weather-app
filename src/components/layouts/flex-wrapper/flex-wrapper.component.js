import styled from 'styled-components';

const FlexWrapper = styled.div`
  display: flex;
  flex-direction: ${props => props.direction || 'initial'};
  justify-content: ${props => props.justify || 'initial'};
  align-items: ${props => props.align || 'initial'};
  gap: ${props => props.gap};
  height: ${props => props.height === 'fill' && '100%'};
`;

export { FlexWrapper };
