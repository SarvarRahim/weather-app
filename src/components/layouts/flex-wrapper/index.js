export * from './flex-wrapper.component';
