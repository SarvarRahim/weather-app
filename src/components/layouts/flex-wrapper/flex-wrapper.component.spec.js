import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { FlexWrapper } from './flex-wrapper.component';

describe('FlexWrapper component', () => {
  test('Render FlexWrapper component', () => {
    const { asFragment } = render(<FlexWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Change flex direction', () => {
    render(<FlexWrapper direction='column' data-testid='flex-wrapper' />);
    const component = screen.getByTestId('flex-wrapper');
    expect(component).toHaveStyleRule('flex-direction', 'column');
  });

  test('Change justify-content', () => {
    render(<FlexWrapper justify='center' data-testid='flex-wrapper' />);
    const component = screen.getByTestId('flex-wrapper');
    expect(component).toHaveStyleRule('justify-content', 'center');
  });

  test('Change align-items', () => {
    render(<FlexWrapper align='flex-end' data-testid='flex-wrapper' />);
    const component = screen.getByTestId('flex-wrapper');
    expect(component).toHaveStyleRule('align-items', 'flex-end');
  });

  test('Change gap', () => {
    render(<FlexWrapper gap='2rem' data-testid='flex-wrapper' />);
    const component = screen.getByTestId('flex-wrapper');
    expect(component).toHaveStyleRule('gap', '2rem');
  });
});
