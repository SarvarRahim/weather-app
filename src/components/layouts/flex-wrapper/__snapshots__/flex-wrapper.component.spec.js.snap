// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexWrapper component Render FlexWrapper component 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: initial;
  -ms-flex-direction: initial;
  flex-direction: initial;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
